dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationEntity.kt

```diff
@@ -62,9 +62,17 @@ data class DataMetaInformationEntity(
      * This function checks these conditions
      */
     fun isDatasetViewableByUser(viewingUser: DatalandAuthentication?): Boolean {
-        return this.qaStatus == QAStatus.Accepted ||
+        return this.qaStatus == QaStatus.Accepted ||
             this.uploaderUserId == viewingUser?.userId ||
-            viewingUser?.roles?.contains(DatalandRealmRole.ROLE_ADMIN) ?: false
+            isDatasetViewableByUserViaRole(viewingUser)
+    }
+
+    private fun isDatasetViewableByUserViaRole(viewingUser: DatalandAuthentication?): Boolean {
+        return viewingUser?.roles?.contains(DatalandRealmRole.ROLE_ADMIN) ?: false ||
+            (
+                viewingUser?.roles?.contains(DatalandRealmRole.ROLE_REVIEWER) ?: false &&
+                    this.qaStatus == QaStatus.Pending
+                )
     }
 
```

Comment:
Does not quite fit in this line but relates to the currentlyActive parameter that is part of this entity. The determination, which dataset is active is now determined by the moment at which the QA of the dataset has been performed rather than when it was uploaded. This has probably always been the case but since QA was instant it didn't matter.

However, now it is possible that we have a dataset that was uploaded in May and one that was uploaded in June. If for some reason the May dataset is QA approved after the June dataset, the May dataset will be the active one. We should clarify with the team if this is what we consider most intuitive.

Reply:
Clarified